src/snowflake/snowpark/mock/util.py

```diff
@@ -164,3 +187,33 @@ def process_string_time_with_fractional_seconds(time: str, fractional_seconds) -
         seconds_part = seconds_part[: min(idx, fractional_seconds)] + seconds_part[idx:]
         ret = f"{time_parts[0]}.{seconds_part}"
     return ret
+
+
+def fix_drift_between_column_sf_type_and_dtype(col: ColumnEmulator):
+    if (
+        isinstance(col.sf_type.datatype, _NumericType)
+        and col.apply(lambda x: x is None).any()
+    ):  # non-object dtype converts None to NaN for numeric columns
+        return col
+    sf_type_to_dtype = {
+        ArrayType: object,
+        BinaryType: object,
+        BooleanType: bool,
+        ByteType: numpy.int8 if not col.sf_type.nullable else "Int8",
+        DateType: object,
+        DecimalType: numpy.int64 if not col.sf_type.nullable else "Int64",
+        DoubleType: numpy.float64,
+        FloatType: numpy.float64,
+        IntegerType: numpy.int64 if not col.sf_type.nullable else "Int64",
+        LongType: numpy.int64 if not col.sf_type.nullable else "Int64",
+        NullType: object,
+        ShortType: numpy.int8 if not col.sf_type.nullable else "Int8",
+        StringType: object,
+        TimestampType: "datetime64[ns]",
```

Comment:
We may add scale to TimestampType in the future. Is it possible that we need to know the exact scale instead of just `TimestampType`?

Reply:
Could you please explain the context more and link any JIRAs? I'm not too sure what you meant by adding scale to TimestampType. Is that a blocker for this PR?

Reply:
I think he means this: https://snowflakecomputing.atlassian.net/browse/SNOW-814253